### PR TITLE
[ENG-2816] Deleting project of for-project registration does not delete node

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Currently draft registrations with deleted nodes show up in the my registrations view, but they should not.

## Changes

- Clean-up queryset to exclude deleted nodes.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify deleted nodes are removed from my registrations page.
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2816

### CODE REVIEW

File: osf/models/user.py [hunk 1]
```diff
@@ -931,9 +931,10 @@ def draft_registrations_active(self):
         """
         Active draft registrations attached to a user (user is a contributor)
         """
+
         return self.draft_registrations.filter(
-            models.Q(registered_node__isnull=True) |
-            models.Q(registered_node__is_deleted=True),
+            (models.Q(registered_node=None) | models.Q(registered_node__is_deleted=True)),
+            branched_from__is_deleted=False,
```
Comment (jwalz) [inline osf/models/user.py:937]: `branched_from__deleted__is_null=True`

**Comment by Johnetordoff** [comment]
@Johnetordoff  Write tests for this.

**Review by jwalz — COMMENTED** [review]
Also, tests

File: api_tests/users/views/test_user_draft_registration_list.py [hunk 2]
```diff
@@ -107,22 +116,19 @@ def test_draft_with_registered_node_does_not_show_up_in_draft_list(
         data = res.json['data']
         assert len(data) == 0
 
-    def test_draft_with_deleted_registered_node_shows_up_in_draft_list(
+    def test_draft_with_deleted_registered_node_does_not_show_up_in_draft_list(
```
Comment (jwalz) [inline api_tests/users/views/test_user_draft_registration_list.py:119]: This shouldn't have changed. Deleting the registered node when it's rejected by admins is part of what makes the draft editable.

**Comment by corbinSanders** [comment]
Looks good!

File: osf/models/user.py [hunk 3]
```diff
@@ -931,9 +931,10 @@ def draft_registrations_active(self):
         """
         Active draft registrations attached to a user (user is a contributor)
         """
+
         return self.draft_registrations.filter(
-            models.Q(registered_node__isnull=True) |
-            models.Q(registered_node__is_deleted=True),
+            (models.Q(registered_node=None) | models.Q(registered_node__deleted__isnull=False)),
```
Comment (jwalz) [inline osf/models/user.py:936]: I think I asked this before, but why the change from `registered_node__isnull=True` to `registered_node=None`?

Comment (Johnetordoff) [reply osf/models/user.py:936]: IIRC It just seemed simpler and more concise.

Comment (jwalz) [reply osf/models/user.py:936]: `__isnull=True` technically has different behavior than `=None`, so please revert to that

File: osf/models/user.py [hunk 4]
```diff
@@ -931,9 +931,10 @@ def draft_registrations_active(self):
         """
         Active draft registrations attached to a user (user is a contributor)
         """
+
         return self.draft_registrations.filter(
-            models.Q(registered_node__isnull=True) |
-            models.Q(registered_node__is_deleted=True),
+            (models.Q(registered_node__isnull=False) | models.Q(registered_node__deleted__isnull=False)),
```
Comment (jwalz) [inline osf/models/user.py:936]: This got flipped somehow in the chaos. We do not want to surface Drafts that have a registered_node (unless the registered_node is also deleted), so this should be `is_null=True`, as it was before.

**Review by jwalz — APPROVED** [review]
